### ISSUE TITLE
[regression]JDBCClient with date on error

### ISSUE THREAD

**Issue by Fyro-Ing** [vert-x3/vertx-jdbc-client#250]
When using JDBCClient (vs JdbcPool) directly occurred error with date on INSERT (probably UPDATE ?)

Using JdbcPool is ok

`io.vertx.serviceproxy.ServiceException: ORA-01843: ce n'est pas un mois valide`

When debugging, all call with JDBCTypeProvider.fromParameter(xx) return JDBCType.OTHER;
Throwing error : 
```JDBCTypeProvider | ParameterMetadata is unsupported bycom.mchange.v2.c3p0.impl.NewProxyPreparedStatement
	at oracle.jdbc.driver.OracleParameterMetaData.checkValidIndex(OracleParameterMetaData.java:219)
	at oracle.jdbc.driver.OracleParameterMetaData.getParameterType(OracleParameterMetaData.java:330)
	at io.vertx.ext.jdbc.impl.actions.JDBCTypeProvider.lambda$fromParameter$0(JDBCTypeProvider.java:27)
	at io.vertx.ext.jdbc.spi.impl.JDBCEncoderImpl.encode(JDBCEncoderImpl.java:27)
	at io.vertx.ext.jdbc.impl.actions.AbstractJDBCAction.fillStatement(AbstractJDBCAction.java:109)
	at io.vertx.ext.jdbc.impl.actions.JDBCUpdate.execute(JDBCUpdate.java:88)
	at io.vertx.ext.jdbc.impl.actions.JDBCUpdate.execute(JDBCUpdate.java:29)
```
This call is ok on SELECT

```
jdbcClient.getConnection(con -> {
 .... 
  con.updateWithParams(sql, map, res -> {})
})
```

Suppose regression from https://github.com/vert-x3/vertx-jdbc-client/pull/235 ?

### Version

4.2.x (ok with 4.1.5)
oracle driver com.oracle.database.jdbc:ojdbc11-production:21.3.0.0

### Context
INSERT INTO MyTable ( DA ) VALUES ( ? )
[2021-11-05T14:35:13.787791287Z"] => date as Instant.now()
DA column is TIMESTAMP(6)

i'll try to do a reproducer test this week-end

**Comment by Fyro-Ing** [comment]
Reproducer with oracle xe : https://github.com/Fyro-Ing/vertx-jdbc-client/commit/6cc0ef4148c6ed3355b46d5c17da3c6f66073676

**Comment by zero88** [comment]
Thank you for your test. I will try it on my side

Please check https://github.com/vert-x3/vertx-jdbc-client/issues/255#issuecomment-976510330

**Comment by zero88** [comment]
`OracleJDBCDriver` is unsupported to get column metadata in a prepared statement/result set.
Then the datatype on the column is fallback to `JDBCType.OTHER`

I think you have to provide customization for `JDBCEncoder` and `JDBCDecoder` based on your data type

**Comment by Fyro-Ing** [comment]
As mentioned, using JdbcPool occured no errors. i've switched to this.

But, on 4.2.x, there is no mention to deprecated or removed usage

**Comment by zero88** [comment]
> But, on 4.2.x, there is no mention to deprecated or removed usage

yep, my mistake. I did not notify the maintainer, just left todo in code about updating the document. 
As of last my patch, I have already included a document about that.

I also added your test (with some modification) in source
https://github.com/vert-x3/vertx-jdbc-client/blob/3518ab0dc553d8836d29a3ea2f88307aeb7d0130/src/test/java/io/vertx/it/OracleTest.java

> As mentioned, using JdbcPool occured no errors. i've switched to this.

I thought you are mentioning `vertx-oracle-client`, coz `JDBCPool` and `JDBCClient` are almost the same behavior when querying/inserting/updating data.
